components/brave_wallet_ui/common/hardware/ledgerjs/sol_ledger_bridge_keyring.ts

```diff
@@ -0,0 +1,116 @@
+  SignHardwareTransactionOperationResult
+} from '../types'
+
+import { hardwareDeviceIdFromAddress } from '../hardwareDeviceIdFromAddress'
+
+export default class SolanaLedgerKeyring implements LedgerSolanaKeyring {
+  private app?: Sol
+  private transport?: Transport
+  private deviceId: string
+
+  coin = (): BraveWallet.CoinType => {
+    return BraveWallet.CoinType.SOL
+  }
+
+  type = (): HardwareVendor => {
+    return LEDGER_HARDWARE_VENDOR
+  }
+
+  getAccounts = async (from: number, to: number): Promise<GetAccountsHardwareOperationResult> => {
+    const unlocked = await this.unlock()
+    if (!unlocked.success || !this.app) {
+      return unlocked
+    }
+    from = (from < 0) ? 0 : from
+    const sol: Sol = this.app
+    const accounts = []
+    for (let i = from; i <= to; i++) {
+      const path = this.getPathForIndex(i)
+      const address = await sol.getAddress(path)
+      accounts.push({
+        address: '',
+        addressBytes: address.address,
+        derivationPath: path,
+        name: this.type(),
+        hardwareVendor: this.type(),
+        deviceId: this.deviceId,
+        coin: this.coin()
+      })
+    }
+    return { success: true, payload: [...accounts] }
+  }
+
+  isUnlocked = (): boolean => {
+    return this.app !== undefined && this.deviceId !== undefined
+  }
+
+  makeApp = async () => {
+    this.transport = await TransportWebHID.create()
+    this.app = new Sol(this.transport)
+  }
+
+  unlock = async (): Promise<HardwareOperationResult> => {
+    if (this.isUnlocked()) {
+      return { success: true }
+    }
+
+    if (!this.app) {
+      await this.makeApp()
+    }
+
+    if (this.app && !this.deviceId) {
+      const sol: Sol = this.app
+      this.transport?.on('disconnect', this.onDisconnected)
+      const zeroPath = this.getPathForIndex(0)
+      const address = (await sol.getAddress(zeroPath)).address
+      this.deviceId = await hardwareDeviceIdFromAddress(address)
+    }
+    return { success: this.isUnlocked() }
+  }
+
+  signPersonalMessage (path: string, address: string, message: string): Promise<SignHardwareMessageOperationResult> {
+    throw new Error('Method not implemented.')
+  }
+
+  signTransaction = async (path: string, rawTxBytes: Buffer): Promise<SignHardwareTransactionOperationResult> => {
+    try {
+      const unlocked = await this.unlock()
+      if (!unlocked.success || !this.app) {
+        return unlocked
+      }
+      const sol: Sol = this.app
+      const signed = await sol.signTransaction(path, rawTxBytes)
+      return { success: true, payload: signed.signature }
+    } catch (e) {
+      return { success: false, error: e.message, code: e.statusCode || e.id || e.name }
+    }
+  }
+
+  private onDisconnected = (e: any) => {
+    if (e.name !== 'DisconnectedDevice') {
+      return
+    }
+    this.app = undefined
+    this.transport = undefined
+  }
+
+  private readonly getPathForIndex = (index: number): string => {
+    return `44'/501'/0'/${index}`
```

Comment:
Those two other paths are not following [bip44](https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki#path-levels) spec
`m / purpose' / coin_type' / account' / change / address_index` because they are not using harden derivation for the child.